storage/store.go

```diff
@@ -65,17 +82,321 @@ type Store interface {
 	) error
 
 	// CollectionByID gets the collection (transaction IDs only) with the given ID.
-	CollectionByID(flowgo.Identifier) (flowgo.LightCollection, error)
+	CollectionByID(ctx context.Context, collectionID flowgo.Identifier) (flowgo.LightCollection, error)
 
 	// TransactionByID gets the transaction with the given ID.
-	TransactionByID(flowgo.Identifier) (flowgo.TransactionBody, error)
+	TransactionByID(ctx context.Context, transactionID flowgo.Identifier) (flowgo.TransactionBody, error)
 
 	// TransactionResultByID gets the transaction result with the given ID.
-	TransactionResultByID(flowgo.Identifier) (types.StorableTransactionResult, error)
+	TransactionResultByID(ctx context.Context, transactionID flowgo.Identifier) (types.StorableTransactionResult, error)
 
 	// LedgerViewByHeight returns a view into the ledger state at a given block.
-	LedgerViewByHeight(blockHeight uint64) *delta.View
+	LedgerViewByHeight(ctx context.Context, blockHeight uint64) *delta.View
 
 	// EventsByHeight returns the events in the block at the given height, optionally filtered by type.
-	EventsByHeight(blockHeight uint64, eventType string) ([]flowgo.Event, error)
+	EventsByHeight(ctx context.Context, blockHeight uint64, eventType string) ([]flowgo.Event, error)
+}
+
+type StorageBackendKeys interface {
+	StorageKey(key string) string
+	LatestBlockKey() []byte
+	BlockHeightKey(height uint64) []byte
+	IdentifierKey(id flowgo.Identifier) []byte
+	EventKey(blockHeight uint64, txIndex, eventIndex uint32, eventType flowgo.EventType) []byte
+}
+
+type StorageBackendData interface {
+	GetBytes(ctx context.Context, store string, key []byte) ([]byte, error)
+	SetBytes(ctx context.Context, store string, key []byte, value []byte) error
+	SetBytesWithVersion(ctx context.Context, store string, key []byte, value []byte, version uint64) error
+	GetBytesAtVersion(ctx context.Context, store string, key []byte, version uint64) ([]byte, error)
+}
```

Comment:
Very nice!!! I really like this refactor.
One note is to avoid exporting this as I don't believe it's needed outside.

NIT: I feel you could rename them a bit tho: remove storage as it's already part of the package name so you have `storage.StorageBackendKeys` -> `storage.BackendKeys` also maybe `storage.KeyGetter` or `storage.KeyGenerator`, but even if just `BackendKeys`is fine I guess just dropping the storage as it's stuttering https://michaelwhatcott.com/go-code-that-stutters/. The same goes for StorageBackendData it could just be BackendData. It would also make sense if this was broken into two interfaces `DataGetter` and `DataSetter`.